packages/widgets/src/definition.ts

```diff
@@ -93,6 +93,7 @@ export type WidgetComponentProps<TKind extends WidgetKind> = WidgetProps<TKind>
 } & {
   boardId: string | undefined; // undefined when in preview mode
   isEditMode: boolean;
+  setOptions: ({ newOptions }: { newOptions: Record<string, unknown> }) => void;
```

Comment:
Change type to `setOptions: ({ newOptions }: { newOptions: Partial<inferOptionsFromDefinition<WidgetOptionsRecordOf<TKind>>> }) => void;`